src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/HttpMethodsProvider.cs

```diff
@@ -73,7 +73,10 @@ public HttpMethodsProvider(IHttpContextAccessor httpContextAccessor)
             _responseWriteMethod = new GlobalMethod
             {
                 Name = "responseWrite",
-                Method = serviceProvider => (Action<string>)(text => httpContextAccessor.HttpContext.Response.WriteAsync(text).GetAwaiter().GetResult())
+                Method = serviceProvider => (Action<string>)(text => {
+                    httpContextAccessor.HttpContext.Items[WorkflowHttpResult.Instance] = WorkflowHttpResult.Instance;
```

Comment:
Why?

Reply:
@Skrypt there is a special type of result **EmptyResult** which represents a result that doesn't do anything and we need to pass information back to the controller when **responseWrite** is used somewhere in script task

we do similar thing in [HttpResponseTask ](https://github.com/OrchardCMS/OrchardCore/blob/a10122828e62aa8f7d56a68111e964fdb6de8ea3/src/OrchardCore.Modules/OrchardCore.Workflows/Http/Activities/HttpResponseTask.cs#L96)

so as long as your tasks or scripting methods are manipulating with response, you need to use EmptyResult, if you will use other possible results, you will get this exception described in related issue

you can read more here https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Workflows/Http/Controllers/HttpWorkflowController.cs

![image](https://user-images.githubusercontent.com/6403130/185576223-810df746-7e66-4545-9938-e4ba89ae4c53.png)

hope that's clear

Reply:
This should be a specific method instead of copying the same code, because in this case the code itself makes it hard to unerstand why this is done. A method name with a summary would make it more explicit.

Reply:
I also wonder if a custom js script should be able to set that, instead of make it automatic.